AppTaskQueue/appscale/taskqueue/push_worker.py

```diff
@@ -71,6 +71,41 @@ def get_wait_time(retries, args):
   return wait_time
 
 
+def update_task(task_name, new_state, retries=3):
+  """ Updates the TaskName entity for a given task.
+
+  Args:
+    task_name: A string specifying the TaskName key.
+    new_state: A string specifying the new task state.
+    retries: An integer specifying how many times to retry the update.
+  """
+  try:
+    task_name_entity = TaskName.get_by_key_name(task_name)
+  except Exception as error:
```

Comment:
Would you mind to limit it to `except db.Error as error:`?
As far as few lines below.